include/envoy/server/instance.h

```diff
@@ -184,6 +185,23 @@ class Instance {
    * @return the flush interval of stats sinks.
    */
   virtual std::chrono::milliseconds statsFlushInterval() const PURE;
+
+  /**
+   * Defines a callback that is run when an admin request completes..
+   * The first parameter to the function is the response-headers, which
+   * will generally have Content-Type and Status fields.
+   */
+  typedef std::function<void(Http::Code, Http::HeaderMap&, absl::string_view)> AdminResponseHandler;
+
+  /**
+   * Admin requests are generally handled by posting to the 'main' thread, so
+   * a callback must be supplied which will be called when the response is handled.
+   *
+   * @param path The admin request URL path as a string.
+   * @param params map of parameter names and values.
+   */
+  virtual void adminRequest(absl::string_view path, const Http::Utility::QueryParams& params,
```

Comment:
The server instance already exposes an `Admin` object. Seems like this isn't needed and should actually get routed through the admin object?

Reply:
Fair point, that shouldn simplify the PR. Thanks.

Reply:
It turns out not to dramatically simplify the PR as I just pushed the mocking boilerplate out of server and into admin :)